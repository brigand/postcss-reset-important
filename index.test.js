var postcss = require('postcss');

var plugin = require('./');

function run(input, output, opts) {
  return postcss([ plugin(opts) ]).process(input, { from: undefined })
    .then(result => {
      expect(result.warnings().length).toBe(0);
      if (output) {
        expect(result.css).toEqual(output);
      } else {
        expect(result.css).toMatchSnapshot();
      }
      return result.css;
    });
}

it('does basic reset', async () => {
  const css = await run('a{ }', null, { });
  expect(css).toMatch(/inherit/);
  expect(css).toMatch(/!important/);
});

it('no reset with reset: false', async () => {
  const css = await run('a{ reset: false }', null, { });
  expect(css).not.toMatch(/inherit/);
});

it('adds important to own classes', async () => {
  const css = await run('a{ color: red; }', null, { });
  expect(css).toMatch(/color: red !important/);
});

