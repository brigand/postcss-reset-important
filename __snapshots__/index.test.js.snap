// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adds important to own classes 1`] = `"a{ margin: 0 !important; padding: 0 !important; border: 0 !important; font-size: 100% !important; font: inherit !important; vertical-align: baseline !important; color: red !important; }"`;

exports[`does basic reset 1`] = `
"a{
    color: inherit !important;
    margin: 0 !important;
    padding: 0 !important;
    border: 0 !important;
    font-size: 100% !important;
    font: inherit !important;
    vertical-align: baseline !important }"
`;

exports[`no reset with reset: false 1`] = `"a{ }"`;
